src/pocketmine/block/Chest.php

```diff
@@ -137,14 +137,17 @@ public function onBreak(Item $item, Player $player = null) : bool{
 	public function onActivate(Item $item, Player $player = null) : bool{
 		if($player instanceof Player){
 			$top = $this->getSide(Vector3::SIDE_UP);
-			if($top->isTransparent() !== true){
+			if(!$top->isTransparent()){
 				return true;
 			}
 
 			$t = $this->getLevel()->getTile($this);
 			$chest = null;
 			if($t instanceof TileChest){
 				$chest = $t;
+				if($chest->getPair() instanceof TileChest and !$chest->getPair()->getBlock()->getSide(Vector3::SIDE_UP)->isTransparent()){
+					return true;
+				}
```

Comment:
I think this would be better positioned with the `lock`ing code further down. A newly-created chest tile may pair with its neighbours.

Reply:
Ooooooooh